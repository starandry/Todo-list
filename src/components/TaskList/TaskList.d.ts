import React from 'react'
interface TaskListProps {
    tasks: {
        id: string
        title: string
        description: string
        completed: boolean
        date: string
        daysSpent: number
        isFrozen: boolean
    }[]
    onToggle: (id: string, completed: boolean, date: string, isFrozen: boolean) => void
    onDelete: (id: string) => void
}
declare const TaskList: React.FC<TaskListProps>
export default TaskList
