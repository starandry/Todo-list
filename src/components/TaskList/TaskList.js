import { jsx as _jsx } from 'react/jsx-runtime'
import Task from '../Task'
import styles from './TaskList.module.scss'
const TaskList = ({ tasks, onToggle, onDelete }) => {
    if (tasks.length === 0) {
        return _jsx('p', { children: 'No tasks for today!' })
    }
    return _jsx('ul', {
        className: styles['task-list'],
        children: tasks.map((task) =>
            _jsx(
                Task,
                {
                    id: task.id,
                    title: task.title,
                    description: task.description,
                    completed: task.completed,
                    date: task.date,
                    daysSpent: task.daysSpent,
                    isFrozen: task.isFrozen,
                    onToggle: () => onToggle(task.id, task.completed, task.date, task.isFrozen),
                    onDelete: () => onDelete(task.id),
                },
                task.id
            )
        ),
    })
}
export default TaskList
