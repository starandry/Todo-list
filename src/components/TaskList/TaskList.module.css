.task-list {
    padding: 0;
    list-style-type: none;
}

/*# sourceMappingURL=TaskList.module.css.map */
