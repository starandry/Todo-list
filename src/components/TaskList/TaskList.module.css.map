{"version":3,"sourceRoot":"","sources":["TaskList.module.scss"],"names":[],"mappings":"AAAA;EACI;EACA","file":"TaskList.module.css"}