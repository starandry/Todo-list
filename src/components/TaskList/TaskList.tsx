import React from 'react';
import Task from '../Task';
import styles from './TaskList.module.scss';

interface TaskListProps {
    tasks: {
        id: string;
        title: string;
        description: string;
        completed: boolean;
        date: string;
        daysSpent: number;
    }[];
    onToggle: (id: string, completed: boolean) => void; // Добавляем completed как параметр
    onDelete: (id: string) => void;
}

const TaskList: React.FC<TaskListProps> = ({ tasks, onToggle, onDelete }) => {
    if (tasks.length === 0) {
        return <p>No tasks for today!</p>;
    }

    return (
        <ul className={styles['task-list']}>
            {tasks.map((task) => (
                <Task
                    key={task.id}
                    id={task.id}
                    title={task.title}
                    description={task.description}
                    completed={task.completed}
                    date={task.date}
                    daysSpent={task.daysSpent}
                    onToggle={() => onToggle(task.id, task.completed)} // Передаем значение completed
                    onDelete={() => onDelete(task.id)}
                />
            ))}
        </ul>
    );
};

export default TaskList;