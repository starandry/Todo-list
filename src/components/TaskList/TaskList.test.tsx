import { render, screen } from '@testing-library/react'
import TaskList from './TaskList'
import '@testing-library/jest-dom'

describe('TaskList Component', () => {
    it('renders "No tasks for today!" when tasks array is empty', () => {
        const mockOnToggle = jest.fn()
        const mockOnDelete = jest.fn()

        // Рендерим TaskList с пустым массивом задач
        const { asFragment } = render(<TaskList tasks={[]} onToggle={mockOnToggle} onDelete={mockOnDelete} />)

        // Проверяем, что текст "No tasks for today!" отображается
        const messageElement = screen.getByText(/No tasks for today!/i)
        expect(messageElement).toBeInTheDocument()

        // Создаём snapshot для пустого списка задач
        expect(asFragment()).toMatchSnapshot()
    })

    it('renders tasks correctly when tasks array is not empty', () => {
        const mockOnToggle = jest.fn()
        const mockOnDelete = jest.fn()

        // Пример массива задач
        const tasks = [
            {
                id: '1',
                title: 'Test Task 1',
                description: 'This is the first test task',
                completed: false,
                date: '2024-10-10',
                daysSpent: 3,
                isFrozen: false,
            },
            {
                id: '2',
                title: 'Test Task 2',
                description: 'This is the second test task',
                completed: true,
                date: '2024-10-09',
                daysSpent: 5,
                isFrozen: true,
            },
        ]

        // Рендерим TaskList с непустым массивом задач
        const { asFragment } = render(<TaskList tasks={tasks} onToggle={mockOnToggle} onDelete={mockOnDelete} />)

        // Проверяем, что заголовки задач отображаются
        const firstTaskTitle = screen.getByText(/Test Task 1/i)
        const secondTaskTitle = screen.getByText(/Test Task 2/i)

        expect(firstTaskTitle).toBeInTheDocument()
        expect(secondTaskTitle).toBeInTheDocument()

        // Создаём snapshot для списка с задачами
        expect(asFragment()).toMatchSnapshot()
    })
})
