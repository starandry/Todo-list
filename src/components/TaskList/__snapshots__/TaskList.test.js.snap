// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskList Component renders "No tasks for today!" when tasks array is empty 1`] = `
<DocumentFragment>
  <p>
    No tasks for today!
  </p>
</DocumentFragment>
`;

exports[`TaskList Component renders tasks correctly when tasks array is not empty 1`] = `
<DocumentFragment>
  <ul
    class="task-list"
  >
    <div
      class="task "
    >
      <h3>
        Test Task 1
      </h3>
      <p>
        This is the first test task
      </p>
      <p
        class="date"
      >
        Created on: 2024-10-10
      </p>
      <p
        class="daysSpent"
      >
        Days spent on this task: 3
      </p>
       
      <div
        class="actions"
      >
        <button>
          Complete
        </button>
        <button>
          Delete
        </button>
      </div>
    </div>
    <div
      class="task completed"
    >
      <h3>
        Test Task 2
      </h3>
      <p>
        This is the second test task
      </p>
      <p
        class="date"
      >
        Created on: 2024-10-09
      </p>
      <p
        class="daysSpent"
      >
        Days spent on this task: 5
      </p>
       
      <div
        class="actions"
      >
        <button>
          Undo
        </button>
        <button>
          Delete
        </button>
      </div>
    </div>
  </ul>
</DocumentFragment>
`;
